### PR TITLE
gnrc_netapi: report errors on dispatch

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This adds the error messages to `gnrc_netapi` in case `gnrc_neterr` is compiled in when a queue of a module is full (`EIO`) or if the `gnrc_netreg` type is not known (`ECANCELED`, should not happen in a properly configured stack).

### Testing procedure
Flood test with `gnrc_sock_udp` with `gnrc_neterr` compiled in. If a queue runs full `gnrc_udp_send()` should return `-EIO`.

### Issues/PRs references
None

### CODE REVIEW

File: sys/net/gnrc/netapi/gnrc_netapi.c [hunk 1]
```diff
@@ -91,20 +93,20 @@ int gnrc_netapi_dispatch(gnrc_nettype_t type, uint32_t demux_ctx,
 
         while (sendto) {
 #if defined(MODULE_GNRC_NETAPI_MBOX) || defined(MODULE_GNRC_NETAPI_CALLBACKS)
-            int release = 0;
+            uint32_t errno = 0;
```
Comment (haukepetersen) [inline sys/net/gnrc/netapi/gnrc_netapi.c]: gcc doesn't like this variable name (for obvious reason) :-)

Comment (miri64) [reply sys/net/gnrc/netapi/gnrc_netapi.c]: Fixed.

**Comment by miri64** [comment]
@haukepetersen ping?

**Comment by benpicco** [comment]
Still needs squashing 😉

**Comment by miri64** [comment]
Oops, done! Thanks for the reminder.

**Pull request merged**